[iOS] Only allow access to the TCC daemon when the GPU process is not enabled

pvollan opened the pull request:
#### ac5c0d67aaf4040d236090f77d11a04bcdff7865
<pre>
[iOS] Only allow access to the TCC daemon when the GPU process is not enabled
<a href="https://bugs.webkit.org/show_bug.cgi?id=262049">https://bugs.webkit.org/show_bug.cgi?id=262049</a>
rdar://115996579

Reviewed by Youenn Fablet.

Block access to the TCC daemon in the WebContent process on iOS when the GPU process is enabled.
When the GPU process is enabled, the code that requires access to TCCD will be running in the
GPU process. There is no longer any need to grant this sandbox extension to the WebContent
process.

* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:
* Source/WebKit/Scripts/process-entitlements.sh:
* Source/WebKit/UIProcess/SpeechRecognitionRemoteRealtimeMediaSourceManager.cpp:
(WebKit::SpeechRecognitionRemoteRealtimeMediaSourceManager::addSource):
(WebKit::SpeechRecognitionRemoteRealtimeMediaSourceManager::removeSource):
* Source/WebKit/UIProcess/SpeechRecognitionRemoteRealtimeMediaSourceManager.h:
* Source/WebKit/WebProcess/Speech/SpeechRecognitionRealtimeMediaSourceManager.cpp:
(WebKit::SpeechRecognitionRealtimeMediaSourceManager::grantSandboxExtensions): Deleted.
(WebKit::SpeechRecognitionRealtimeMediaSourceManager::revokeSandboxExtensions): Deleted.
* Source/WebKit/WebProcess/Speech/SpeechRecognitionRealtimeMediaSourceManager.h:
* Source/WebKit/WebProcess/Speech/SpeechRecognitionRealtimeMediaSourceManager.messages.in:
* Source/WebKit/WebProcess/com.apple.WebProcess.sb.in:

Canonical link: <a href="https://commits.webkit.org/268586@main">https://commits.webkit.org/268586@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8b57e42ba8f6a89cbc91ab25a81804a3c0c57ae6

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/19992 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/20417 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/21040 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/21887 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/18671 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/23667 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/20568 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/20171 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/20212 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/20158 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/17382 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/22739 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/20178 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/17328 "Passed tests") | | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/24437 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/18405 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/18354 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/22426 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/18944 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/16071 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/18125 "Built successfully") | | | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/19/builds/4824 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/22476 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/2468 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/18770 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:
```diff
@@ -930,7 +930,9 @@
             "com.apple.mobilegestalt.xpc"
+#if !ENABLE(GPU_PROCESS_BY_DEFAULT)
             "com.apple.tccd"
+#endif
```
Can you also remove the `plistbuddy` commands from `process-entitlements.sh`?

brentfulgham replied:
It looks like Speech Recognition might want to use the microphone in the WebContent process. This should probably be moved to the GPU process (and perhaps already has been).

See `SpeechRecognitionRealtimeMediaSourceManager` for details.

pvollan replied:
> Can you also remove the `plistbuddy` commands from `process-entitlements.sh`?

Ah, good point, will fix!

Thanks for reviewing!

pvollan replied:
> It looks like Speech Recognition might want to use the microphone in the WebContent process. This should probably be moved to the GPU process (and perhaps already has been).
> 
> See `SpeechRecognitionRealtimeMediaSourceManager` for details.

That is a good point. I have removed the sandbox extensions for this case in the latest patch.

youennf commented on Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:
```diff
@@ -930,7 +930,9 @@
             "com.apple.iphone.axserver-systemwide"
             "com.apple.mobileassetd.v2"
             "com.apple.mobilegestalt.xpc"
+#if !ENABLE(GPU_PROCESS_BY_DEFAULT)
```
Should we do the same for macOS where capture is either UIProcess or GPUProcess?

youennf reviewed (approved):
LGTM once bots are happy.
I wonder whether we should not do the same for microphone/camera in WebProcess sandboxes as well.

pvollan committed:
[iOS] Only allow access to the TCC daemon when the GPU process is not enabled
https://bugs.webkit.org/show_bug.cgi?id=262049
rdar://115996579

Reviewed by Youenn Fablet.

Block access to the TCC daemon in the WebContent process on iOS when the GPU process is enabled.
When the GPU process is enabled, the code that requires access to TCCD will be running in the
GPU process. There is no longer any need to grant this sandbox extension to the WebContent
process.

* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:
* Source/WebKit/Scripts/process-entitlements.sh:
* Source/WebKit/UIProcess/SpeechRecognitionRemoteRealtimeMediaSourceManager.cpp:
(WebKit::SpeechRecognitionRemoteRealtimeMediaSourceManager::addSource):
(WebKit::SpeechRecognitionRemoteRealtimeMediaSourceManager::removeSource):
* Source/WebKit/UIProcess/SpeechRecognitionRemoteRealtimeMediaSourceManager.h:
* Source/WebKit/WebProcess/Speech/SpeechRecognitionRealtimeMediaSourceManager.cpp:
(WebKit::SpeechRecognitionRealtimeMediaSourceManager::grantSandboxExtensions): Deleted.
(WebKit::SpeechRecognitionRealtimeMediaSourceManager::revokeSandboxExtensions): Deleted.
* Source/WebKit/WebProcess/Speech/SpeechRecognitionRealtimeMediaSourceManager.h:
* Source/WebKit/WebProcess/Speech/SpeechRecognitionRealtimeMediaSourceManager.messages.in:
* Source/WebKit/WebProcess/com.apple.WebProcess.sb.in:

Canonical link: https://commits.webkit.org/268586@main
